File: tldjfj123/Week06/전화번호목록/Solution(88.3).py
def solution(phone_book):
    phone_book.sort()
    table = dict()
    
    for p in phone_book :
        if not table.keys() :
            table[p] = 1
        else :
            for key in table.keys() :
                if key == p[:len(key)] :
                    return False
            else :
                table[p] = 1
    else :
        return True
        
    
    
    

File: tldjfj123/Week06/전화번호목록/Solution(88.3).java
import java.util.Arrays;
import java.util.HashMap;

public class Solution {

    public static boolean solution(String[] phoneBook) {
        Arrays.sort(phoneBook);
        HashMap<String, Integer> table = new HashMap<>();

        for (String p : phoneBook) {
            if (table.isEmpty()) {
                table.put(p, 1);
            } else {
                boolean foundPrefix = false;
                for (String key : table.keySet()) {
                    if (p.startsWith(key)) {
                        foundPrefix = true;
                        break;
                    }
                }
                if (foundPrefix) {
                    return false;
                } else {
                    table.put(p, 1);
                }
            }
        }
        return true;
    }
}
